### ISSUE TITLE
fold change information

### ISSUE THREAD

**Issue by ghost** [scverse/scanpy#673]
Hi, I would like to know why the exp1 function was used to calculate the fold change:

```py
foldchanges = (np.expm1(means[imask]) + 1e-9) / (np.expm1(mean_rest) + 1e-9)`
```

instead of

```py
foldchanges = means[imask] + 1e-9 / mean_rest + 1e-9
```

Thank you.

**Comment by ghost** [comment]
Hi, is it possible that the you used the expm1 function to to correct the fold changes?

**Comment by LuckyMD** [comment]
@Koncopd @a-munoz-rojas was it one of you that introduced fold changes info into `sc.tl.rank_genes_groups()` that could answer this?

**Comment by ghost** [comment]
I think that `sc.pp.log1p(adata)` is the important step to execute before to calculate the fold changes because in this case the fold change would be calculated in this way:
`foldchanges = (np.expm1(mean(log1p())) + 1e-9) / (np.expm1(mean(log1p())) + 1e-9)`, is it right?

**Comment by LuckyMD** [comment]
Yeah, so it appears that the function expects log-transformed data. Otherwise the use of `np.expm1()` doesn't make sense. If you don't log transform your data, you would get the wrong values. It might be worth either auto-detecting whether data was log-transformed or not, or adding it as a parameter.

**Comment by a-munoz-rojas** [comment]
Hi! Sorry for the very late reply! But yes, this function is assuming the data is log-transformed before ranking genes. That's the typical workflow. It originally had a parameter to check for this, but then we decided to simplify and remove it (#519). There was some brief discussions here about adding an attribute when pp.log1p is run to handle non-transformed data, but I don't think was ever implemented. Might be worth revisiting though

**Comment by flying-sheep** [comment]
seems answered
